sdk_contrib/aws_v1_support/src/main/java/io/opentelemetry/sdk/contrib/trace/aws/propagation/AWSXRayPropagator.java

```diff
@@ -0,0 +1,238 @@
+ *     .addHttpTextFormat(new AWSXRayPropagator()).build());
+ * </pre>
+ */
+public class AWSXRayPropagator implements HttpTextFormat {
+
+  private static final Logger logger = Logger.getLogger(AWSXRayPropagator.class.getName());
+
+  static final String TRACE_HEADER_KEY = "X-Amzn-Trace-Id";
+  private static final char TRACE_HEADER_DELIMITER = ';';
+  private static final char KV_DELIMITER = '=';
+
+  private static final String TRACE_ID_KEY = "Root";
+  private static final String TRACE_ID_VERSION = "1";
+  private static final char TRACE_ID_DELIMITER = '-';
+  private static final int TRACE_ID_FIRST_PART_LENGTH = 8;
+  private static final int TRACE_ID_SECOND_PART_LENGTH = 24;
+
+  private static final String PARENT_ID_KEY = "Parent";
+  private static final int PARENT_ID_LENGTH = 16;
+
+  private static final String SAMPLED_FLAG_KEY = "Sampled";
+  private static final int SAMPLED_FLAG_LENGTH = 1;
+  private static final char IS_SAMPLED = '1';
+  private static final char NOT_SAMPLED = '0';
+
+  private static final List<String> FIELDS = Collections.singletonList(TRACE_HEADER_KEY);
+
+  @Override
+  public List<String> fields() {
+    return FIELDS;
+  }
+
+  @Override
+  public <C> void inject(Context context, @Nullable C carrier, Setter<C> setter) {
+    Objects.requireNonNull(context, "context");
+    Objects.requireNonNull(setter, "setter");
+
+    Span span = TracingContextUtils.getSpanWithoutDefault(context);
+    if (span == null) {
+      return;
+    }
+
+    SpanContext spanContext = span.getContext();
+
+    String otTraceId = spanContext.getTraceId().toLowerBase16();
+    String xrayTraceId = TRACE_ID_VERSION + TRACE_ID_DELIMITER +
+        otTraceId.substring(0, TRACE_ID_FIRST_PART_LENGTH) + TRACE_ID_DELIMITER +
+        otTraceId.substring(TRACE_ID_FIRST_PART_LENGTH);
+    String parentId = spanContext.getSpanId().toLowerBase16();
+    char samplingFlag = spanContext.getTraceFlags().isSampled() ? IS_SAMPLED : NOT_SAMPLED;
+
+    String traceHeader = TRACE_ID_KEY + KV_DELIMITER + xrayTraceId + TRACE_HEADER_DELIMITER +
+        PARENT_ID_KEY + KV_DELIMITER + parentId + TRACE_HEADER_DELIMITER +
+        SAMPLED_FLAG_KEY + KV_DELIMITER + samplingFlag;
+    setter.set(carrier, TRACE_HEADER_KEY, traceHeader);
+  }
+
+  @Override
+  public <C> Context extract(Context context, C carrier, Getter<C> getter) {
+    checkNotNull(carrier, "carrier");
+    checkNotNull(getter, "getter");
+
+    SpanContext spanContext = getSpanContextFromHeader(carrier, getter);
+    return TracingContextUtils.withSpan(DefaultSpan.create(spanContext), context);
+  }
+
+  @SuppressWarnings("StringSplitter")
+  private static <C> SpanContext getSpanContextFromHeader(C carrier, Getter<C> getter) {
+    try {
+      String traceHeader = getter.get(carrier, TRACE_HEADER_KEY);
+      if (StringUtils.isNullOrEmpty(traceHeader)) {
+        return SpanContext.getInvalid();
+      }
+
+      String[] parts = traceHeader.split(String.valueOf(TRACE_HEADER_DELIMITER));
```

Comment:
We want our instrumentation to have as low an impact on user app's performance as possible so we should take care in optimizing especially parse logic. 

This is probably the most optimized version

https://github.com/openzipkin/zipkin-aws/blob/master/brave-propagation-aws/src/main/java/brave/propagation/aws/AWSExtractor.java#L72

Even if that's too complex, we can still get a lot of improvement by avoiding `.split` in favor of `indexOf` and `substring`, I don't think we need to use `split` at all here (the warning is valid :) )

Reply:
JDK has a specialized optimization on splitting by single character, which doesn't use regex. See [SO post](https://stackoverflow.com/questions/11001330/java-split-string-performances) and [JDK implementation](http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/lang/String.java#l2324). Hence I don't think changing to the extractor or `substring` will make noticeable performance improvement that worths the trade of readability.

Reply:
It's not about the regex but the allocation of an array is still significant, especially since we don't need it. On the flip side, the change in readability also shouldn't be significant.

Reply:
Ok that makes sense. Fixing in rev3